### ISSUE TITLE
Drag and drop tabs sometimes results in wrong order

### ISSUE THREAD

**Issue by ludasmokva** [Croydon/vertical-tabs-reloaded#162]
Add-on version: 0.9.4
Firefox version: 57.0.1 64-bit
OS + OS-Version: Win 7
Other add-ons (complete list):  
  Decentraleyes
  Disconnect
  Font Finder
  Google search link fix
  Nimbus Screen Capture
  No Script
  Turn Off the Lights
  uBlock Origin
  Vertical Tabs

Ok... these are simply some weird things I've noticed since FF57 and I wanted to let you know. 
 1.  The right click on the vertical tabs has become very limited (only 4). Mostly I miss my 'close tabs to the right' option as I tend to open a whole bunch of them and find it tedious to close each one individually. The thing is, this right-click option is still available on the horizontal tabs (12 options).... but a different issue makes this unreliable. (see #2)
2. The horizontal tabs and vertical tabs are not aligned. So if I move a vertical tab, it does not reflect in horizontal tabs and vice-versa (note: I've actually noticed instances where moving one results in the other one being moved as well but this is so rare that I have not been able to figure out what the trigger is.) In any case, this is not a real biggie as I don't need to use both, BUT, that is what makes issue #1 annoying. Cannot use horizontal option 'delete tabs to the right' if I have moved around any of my vertical tabs. But using vertical tabs forces me to delete/close tabs one at a time.

This may end up being resolved when FF fixes the horizontal tab issue,  but please double check why the same options do not appear on the right-click and make certain those don't disappear when the horizontal tabs disappear.
Thanks and keep up the great work!

**Comment by Croydon** [comment]
Please use the search before reporting issues.
1. is a duplicate of #120

Not sure about 2. I can't reproduce this right now.

**Comment by Croydon** [comment]
@ludasmokva Could you please re-evaluate the situation with 0.10.0?

**Comment by ludasmokva** [comment]
Retested... briefly. Issue still remains.
So here is what I did.
Assume 10 tabs open.
If I clicked on the horizontal tab#3 to open link in new tab, on the horizontal bar, new tab is in position 4 while in vertical tab it is in position 11 (very bottom).
I also tried moving a few tabs around, and 95% of the times it works. But some don't. 
The end tabs tabs won't move at all. For instance, I have difficulty moving vertical tab 10 to position 11 and vice versa. They simply won't move. The middle tabs seem to move ok. 
But, like I said, a particular tab is in spot 4 on the horizontal bar, but spot 10 on the vertical side. So not sure if that plays into it. 
Let me know if I can be of any further help.
This is weird. 
Maybe it's because you don't have the option to open tab next to original tab? So the FF default only affects horizontal but not the vertical order? And that in turn is playing havoc on this order thing.

**Comment by Croydon** [comment]
@ludasmokva Do I understand your comment here https://github.com/Croydon/vertical-tabs-reloaded/issues/159#issuecomment-385208602 correctly, that this is solved for you as well?

**Comment by ludasmokva** [comment]
Yup. It is solved.
The icons are working/responding.
Moving tabs on vertical section moves them on horizontal and vice-versa.

Sorry about going on and on before... but I wanted to give you some
background in case you are able to figure out what had happened and how it
may have resolved itself and perhaps help others.
It is the craziest thing and  I wish I knew what I did.
It did not fix itself on any one action that I know of.
Resetting vertical bars showed no immediate reaction.
Restarting FF showed no immediate reaction.
Small adjustments to FF options showed no immediate reaction
Pressing combinations of ALT-SHFT-CTRL + various keys.... showed no
immediate reaction.

What happened was that at some point while I was working on fixing the
language issue, I maximized my minimized FF to test in the textbox if the
accented E was still showing up instead of the question mark. And when I
maxed the window, suddenly the vertical bars were there.
I tested the icons simply out of curiosity, and much to my surprise, they
were working!
I started moving tabs around to see if that was still and issue, but they
worked great.
(And if you're interested... the language issue was fixed as well.)

It is all very weird and I tend to think that it was less about your
program but rather the way FF was integrating your program. Or maybe how
another add-on was acting up.
One thing that may or may not have had an impact is NoScript. When I was on
the add-on page for VTB trying to determine why the tabs were not
displaying, it seemed like NoScript was blocking something so I simply
allowed that page. I did not notice any immediate response, but maybe it
took a while to take effect.

Don't know if this is worth your looking into it any further, but figured
more data is always more helpful than less.
Sorry if I got too wordy and confused the issue.
Let me know if you would like me to test out anything else or if you would
still like me install the experimental version.
You have been completely awesome and I would be very happy to help in any
way.... even if it temporarily screws something up! :D

**Comment by Croydon** [comment]
> Pressing combinations of ALT-SHFT-CTRL + various keys.... showed no immediate reaction.

What did you expect to happen from these hotkeys?

> Don't know if this is worth your looking into it any further

Since your situation seems to be very hard to reproduce and the issues seems to be fixed and also nobody else came forward with these problems I would rather move on here :)

> Let me know if you would like me to test out anything else or if you would still like me install the experimental version.

Shouldn't be neccessary for the moment. Let me know if you have any further issues.

**Comment by ludasmokva** [comment]
(ME) Pressing combinations of ALT-SHFT-CTRL + various keys.... showed no
immediate reaction.

 (YOU) What did you expect to happen from these hotkeys?

Simply meant that

Shift+CTRL+V = display VTB tabs: did not work

LEFT Alt + Shift = switching btw languages: no change when pressed - that
was the weird issue that started this whole thing.

Thanks again! Hopefully no more issues.
